### PR TITLE
Fix NPC followers being blind after waking up

### LINKED ISSUE #43079: NPC followers are blind after waking up from sleep
# Describe the bug

After going to sleep and waking up naturally or through the dialogue option, NPC followers appear to be blind.  If you are not in an adjacent tile, their icon will indicate they don't see you, the "V" menu lists them as "unaware of your presence", and they will not move to engage visible enemies.  If you are fighting an enemy that is not adjacent to them, they will make comments about hearing things like "that sounds bad", etc.  They will no longer make comments about spotting distant enemies.  Looking at their character screens shows nothing out of the ordinary.

# Steps To Reproduce
1. Recruit an NPC follower
2. Wait for them to fall asleep
3. Wait for them to wake up or tell them to wake up via "C"hat command
4. They are now apparently blind.

# Expected behavior

Followers shouldn't be blinded by sleeping.

# Screenshots

# Versions and configuration
OS: Windows 10
Game Version: 0.E-4878-gf4440a4 
Ingame Language: English
Mods loaded: disable npc needs,  sees-player-icon-retrodays

# Additional context

I was able to do a workaround by crafting blindfolds, telling follower "I want you to use this item" and selecting the blindfold, then removing said blindfold through the "sort armor" menu for followers.  This seems to make them see again until the next time they go to sleep.

### PR DESCRIPTION
**Pull request by raito133**
#### Summary
Bugfixes "Fix NPC followers being blind after waking up"

#### Purpose of change
Fixes #43079. NPC followers wake up blind (they are aware of your presence only when adjacent to them) because in `Character::wake_up()` `recalc_sight_limits()` is called before `effect_sleep` is removed. `recalc_sight_limits()` sets `sight_max` to 0 like so:
```cpp 
if( is_blind() || ( in_sleep_state() && !has_trait( trait_SEESLEEP ) ) ||
        has_effect( effect_narcosis ) ) {
        sight_max = 0;
```
The player character doesn't wake up blind because  `recalc_sight_limits()` is called frequently enough for them, which is not the case for NPCs. Blinded followers will get their sight again when ordered to wear something or after reloading the game, because this calls  `recalc_sight_limits()` for them. 

#### Describe the solution

When waking up, do the recalculation only if this character is the player. NPC followers don't lose sight when falling asleep  anyway.

#### Describe alternatives you've considered

Change  `recalc_sight_limits()` so that it doesn't change `sight_max` when having `effect_sleep` for followers.
Call `recalc_sight_limits()` when the `effect_sleep` is removed. `effect_sleep` is removed during `Creature::process_effects()` and this function is too abstract for this. 

#### Testing

Forced NPC followers to sleep and then to wake up. Went to sleep as the player character.

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -9143,7 +9143,9 @@ void Character::wake_up()
     if( has_effect( effect_alarm_clock ) ) {
         get_effect( effect_alarm_clock ).set_duration( 0_turns );
     }
-    recalc_sight_limits();
+    if ( is_player() ) {
```
Comment (BrettDong) [inline src/character.cpp:9146]:
```suggestion
    if( is_player() ) {
```

**Comment by BrettDong** [comment]
I woke up an NPC follower through chat menu but the NPC is still blind.

**Comment by BrettDong** [comment]
I think the proper fix is
```diff
diff --git a/src/npctalk_funcs.cpp b/src/npctalk_funcs.cpp
index be2aa274fa..0b25b660d2 100644
--- a/src/npctalk_funcs.cpp
+++ b/src/npctalk_funcs.cpp
@@ -434,6 +434,7 @@ void talk_function::wake_up( npc &p )
     p.remove_effect( effect_lying_down );
     p.remove_effect( effect_npc_suspend );
     p.remove_effect( effect_sleep );
+    p.recalc_sight_limits();
     // TODO: Get mad at player for waking us up unless we're in danger
 }
```

**Comment by raito133** [comment]
> 
> 
> I woke up an NPC follower through chat menu but the NPC is still blind.

I can't reproduce this. NPCs don't calculate their `sight_max` when going to sleep or when asleep. Unless they are asleep already when loading the game or they mutate/get bionics/wear an item/mount a creature during their sleep. Which makes me think that this will be a better fix:
revert this:
```diff
-    if( is_player() ) {
-        recalc_sight_limits();
-    }
+    recalc_sight_limits();
```
and add this:
```diff
-    if( is_blind() || ( in_sleep_state() && !has_trait( trait_SEESLEEP ) ) ||
+    if( is_blind() || ( in_sleep_state() && !has_trait( trait_SEESLEEP ) && is_player() ) ||
         has_effect( effect_narcosis ) ) {
         sight_max = 0;
     }
  }
```

**Review by BrettDong — APPROVED** [review]

**Pull request merged**